jcl/src/java.base/share/classes/java/lang/invoke/FilterArgumentsWithCombinerHandle.java

```diff
@@ -0,0 +1,108 @@
+/*[INCLUDE-IF Java12]*/
+/*******************************************************************************
+ * Copyright (c) 2018, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be madpe available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package java.lang.invoke;
+
+final class FilterArgumentsWithCombinerHandle extends MethodHandle {
+    protected final MethodHandle handle;
+    protected final MethodHandle combiner;
+    private final int filterPosition;
+    private final int[] argumentIndices;
+
+    FilterArgumentsWithCombinerHandle(MethodHandle handle, int filterPosition, MethodHandle combiner, int... argumentIndices) {
```

Comment:
Can you add a comment either to the constructor or the class that describes the algorithm?  Is it combine then filter?  or filter than combine?